### PR TITLE
Add format command.

### PR DESCRIPTION
**Pull request by andreabedini**
This command reads and re-writes (in-place) the metadata files. When applied to CHaP, the result is

```diff
--- a/_sources/vector-map/1.0.1.0/meta.toml
+++ b/_sources/vector-map/1.0.1.0/meta.toml
@@ -1,3 +1,6 @@
-timestamp = 2023-04-05T21:13:03Z
-github = { repo = "input-output-hk/cardano-ledger", rev = "f10f06f6ab96b5ee52a28ccc45b41a592efde4b7" }
-subdir = 'libs/vector-map'
+subdir = "libs/vector-map"
+timestamp = 2023-04-05T21:13:03+00:00
+
+[github]
+  repo = "input-output-hk/cardano-ledger"
+  rev = "f10f06f6ab96b5ee52a28ccc45b41a592efde4b7"
```

It would be better if CHaP's `add-from-github.sh` would use the same formatting but it would be even better to put that functionality inside foliage itself.

### CODE REVIEW

File: app/Foliage/CmdLint.hs [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    <$> strOption
+      ( long "input-directory"
+          <> metavar "INPUT"
+          <> help "Repository input directory"
+          <> showDefault
+          <> value "_sources"
+      )
+
+cmdLint :: LintOptions -> IO ()
+cmdLint buildOptions = do
```
Comment (michaelpj) [inline app/Foliage/CmdLint.hs:23]:
```suggestion
cmdLint lintOptions = do
```

File: app/Foliage/CmdLint.hs [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  shake opts $
+    do
+      phony "lintAction" (lintAction buildOptions)
+      want ["lintAction"]
+  where
+    cacheDir = "_cache"
+    opts = shakeOptions {shakeFiles = cacheDir, shakeVerbosity = Verbose}
```
Comment (michaelpj) [inline app/Foliage/CmdLint.hs:30]: Should we share this stuff? Seems like the shake options should pretty much always be the same?

**Review by michaelpj — COMMENTED** [review]
I wouldn't expect a "lint" command to modify my source tree by default. It could have a `--fix` flag or something that tells it to actually try to fix problems.

**Comment by andreabedini** [comment]
> I wouldn't expect a "lint" command to modify my source tree by default. It could have a --fix flag or something that tells it to actually try to fix problems.

Yes, you are right. Also, what this PR introduces is more a `format` than a `lint`.

File: app/Foliage/CmdLint.hs (hunk 2 above)
Comment (andreabedini) [reply app/Foliage/CmdLint.hs]: We should. Also we should wire the verbosity to a command line flag. Also I don't like how shake prints messages.

**Comment by andreabedini** [comment]
I believe this is not the direction we want to go.

**Pull request closed without merging**